src/routes/api/user/index.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  FastifyPluginAsyncTypebox,
+  Type
+} from '@fastify/type-provider-typebox'
+import { Auth } from '../../../schemas/auth.js'
+import { UpdateCredentialsSchema } from '../../../schemas/users.js'
+
+const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
+  fastify.put(
+    '/update-password',
+    {
+      schema: {
+        body: UpdateCredentialsSchema,
+        response: {
+          200: Type.Object({
+            message: Type.String()
+          }),
+          401: Type.Object({
+            message: Type.String()
+          })
+        },
+        tags: ['User']
+      }
+    },
+    async function (request, reply) {
+      const { newPassword, currentPassword } = request.body
+      const username = request.session.user.username
+
+      try {
+        const user = await fastify.knex<Auth>('users')
+          .select('username', 'password')
+          .where({ username })
+          .first()
+
+        if (user) {
```

Comment:
```suggestion
        if (!user) {
```
Prefer early return, if there is no user, the program can stop.